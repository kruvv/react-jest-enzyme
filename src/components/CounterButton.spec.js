import { shallow } from 'enzyme';
import React, { Component } from 'react';
import CounterButton from './CounterButton';


const setUp = (props) => shallow(<CounterButton {...props} />);

describe("<CountButton />", () => {
    let component;
    let instance;

    beforeEach(() => {
        component = setUp();
        instance = component.instance();
    });

    it("should correctly render component CountButton", () => {
        expect(component).toMatchSnapshot();
    });
    describe("CountButton handlers", () => {
        it("should changecount value to plus", () => {
            const btn = component.find(".plusOneBtn");
            btn.simulate("click");
            // expect(component).toMatchSnapshot(); // можно использовать, но не очень удобно для больших компонентов
            expect(component.state().count).toBe(1) // change state
        });

        // Variant #1
        it("should reset count value to 10", () => {
            const btn = component.find(".resetBtn");
            btn.simulate("click");
            expect(component.state().count).toBe(10) // change state
        });
        // Variant #2
        it("should reset count value to custom value", () => {
            instance.handleReset(20)
            expect(component.state().count).toBe(20) // change state
        });
    });
})
