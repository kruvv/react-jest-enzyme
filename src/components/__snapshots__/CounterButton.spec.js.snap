// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountButton /> should correctly render component CountButton 1`] = `
<div>
  <div>
    0
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;
